gulpfile.js

```diff
@@ -2,26 +2,65 @@
 
 (function (pkg, gulp,
            jshint, jscs, stylish, mocha, connect, less, requirejs, uglify, replace, rename, header, awspublish,
-           aws, yargs, rimraf, mkdirp, recursive) {
+           aws, yargs, fs, path, rimraf, mkdirp, recursive) {
     'use strict';
 
-    var environment = yargs.argv.env || 'development',
-        environmentPrefix = {
-            staging: 'staging-',
-            uat: 'uat-',
-            production: ''
-        },
-        headerTemplate =
-            '/**\n' +
-            ' * ${pkg.name}\n' +
-            ' * ${pkg.description}\n' +
-            ' * \n' +
-            ' * @version ${pkg.version}\n' +
-            ' * @date ' + (new Date()).toISOString() + '\n' +
-            ' * @environment ' + environment + '\n' +
-            ' * @link ${pkg.homepage}\n' +
-            ' * @license ${pkg.license}\n' +
-            ' */\n\n';
+    var environment, environmentPrefix, headerTemplate, autodiscoverTests;
+
+    environment = yargs.argv.env || 'development';
+
+    environmentPrefix = {
+        staging: 'staging-',
+        uat: 'uat-',
+        production: ''
+    };
+
+    headerTemplate =
+        '/**\n' +
+        ' * ${pkg.name}\n' +
+        ' * ${pkg.description}\n' +
+        ' * \n' +
+        ' * @version ${pkg.version}\n' +
+        ' * @date ' + (new Date()).toISOString() + '\n' +
+        ' * @environment ' + environment + '\n' +
+        ' * @link ${pkg.homepage}\n' +
+        ' * @license ${pkg.license}\n' +
+        ' */\n\n';
+
+    autodiscoverTests = function (callback) {
```

Comment:
This code is a bit ugly, but I couldn't make it much better with `async` so I left it like this and one less dependency.